src/osd/PeeringState.h

```diff
@@ -1765,6 +1765,17 @@ class PeeringState : public MissingLoc::MappingInfo {
     std::optional<eversion_t> trim_to,
     std::optional<eversion_t> roll_forward_to);
 
+  void append_log_with_trim_to_updated(
+    const std::vector<pg_log_entry_t>& logv,
```

Comment:
a little bit off-topic, can we change `PeeringState::append_log()` so it accepts a rvalue reference for `logv` ? i checked all its callers, none of them are expecting the availability of `logv` after passing it to `append_log()`. the entry in `logv` will be appended to `pg_log_t::log`. and `pg_log_entry_t` is not a lightweight object in the sense of its size of the overhead of copying it, so i think it'd be beneficial to classic osd as well.

@athanatos @rzarzynski @xxhdx1985126 what do you think?

Reply:
:+1: